### PR TITLE
Orleans 3.0 + AspNet Core 3.0 Endpoints

### PR DESCRIPTION
**Pull request by galvesribeiro**
- [x] Endpoint Routing to grains based on attributes
- [x] Authentication / Authorization
- [ ] Minimal unit tests
- [x] JSON (System.Text.Json)
- [x] Protobuf (protobuf-net)
- [x] XML
- [x] Forms

### CODE REVIEW

**Comment by galvesribeiro** [comment]
Almost there...

File: src/Orleans.Http.MediaTypes.Protobuf/PipeReaderStream.cs [hunk 1]
```diff
@@ -0,0 +1,240 @@
+namespace Orleans.Http.MediaTypes.Protobuf
+{
+    using System;
+    using System.Buffers;
+    using System.IO;
+    using System.IO.Pipelines;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal class PipeStream : Stream
```
Comment (davidfowl) [inline src/Orleans.Http.MediaTypes.Protobuf/PipeReaderStream.cs]: Why do you need this? Just call AsStream on the PipeWriter/PipeReader

File: src/Orleans.Http.MediaTypes.Protobuf/ProtobufMediaTypeHandler.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace Orleans.Http.MediaTypes.Protobuf
+{
+    internal sealed class ProtobufMediaTypeHandler : IMediaTypeHandler
+    {
+        public string MediaType => "application/protobuf";
+
+        public ValueTask<object> Deserialize(PipeReader reader, Type type, CancellationToken cancellationToken)
+        {
+            using var stream = new PipeStream(reader, false);
```
Comment (davidfowl) [inline src/Orleans.Http.MediaTypes.Protobuf/ProtobufMediaTypeHandler.cs]: `reader.AsStream()`

File: src/Orleans.Http.MediaTypes.Protobuf/PipeReaderStream.cs (hunk 1 above)
Comment (galvesribeiro) [reply src/Orleans.Http.MediaTypes.Protobuf/PipeReaderStream.cs]: Indeed. I don't remember now why I had an issue. Just ran the tests and it worked...

File: src/Orleans.Http/HostingExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                    {
+                        route.RequireAuthorization(
+                            new ASPNetAuthorizeAttribute(authorizeAttr.Policy)
+                            {
+                                Roles = authorizeAttr.Roles,
+                                AuthenticationSchemes = authorizeAttr.AuthenticationSchemes
+                            });
+                    }
+
+                    logger.LogInformation($"[{httpMethod}] [{grainType.FullName}.{method.Name}] -> {routePattern.RawText}.");
```
Comment (ReubenBond) [inline src/Orleans.Http/HostingExtensions.cs]: Ideally this uses structured logging, but that can come later.

**Comment by galvesribeiro** [comment]
Ok, first round is in. Tests and reflection-removal later on...

**Pull request merged**